docs/src/pages/components/data-grid/editing/editing.md

```diff
@@ -5,7 +5,123 @@ components: DataGrid, XGrid
 
 # Data Grid - Editing
 
-<p class="description">Data Grid will provide full support to create, read, update, and delete operations (CRUD).</p>
+<p class="description">The data grid has built-in edit capabilities that you can customize to your needs.</p>
+
+## Cell editing
+
+Cell editing allows to edit the value of one cell at a time.
+
+- set `editable` attribute of the column definition `GridColDef` object to `true` to allow editing cells of this column.
+
+```tsx
+<DataGrid columns={[{ field: 'name', editable: true }]} />
+```
+
+### Start Editing
+
+If a column is editable and if a cell has focus.
+
+- Press <kbd class="key">Enter</kbd> to switch to edit mode
+- Press <kbd class="key">Delete</kbd> to delete the value
+- Type any alpha key to start editing
+- Double click to switch to edit mode
+
+{{"demo": "pages/components/data-grid/editing/BasicEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Stop Editing
+
+- <kbd class="key">Escape</kbd> to rollback changes.
+- <kbd class="key">Tab</kbd> to save and goes to next cell.
+- <kbd class="key">Enter</kbd> to save and goes to next cell below.
+- Click outside the cell
+
+### Control cell editability
+
+In addition to the `editable` flag on columns, control which cell is editable using `isCellEditable` prop.
+
+_In this demo, `Name` is not editable, and only the rows with an even `Age` value are editable.
+We applied a green background for a better visibility._
+
+{{"demo": "pages/components/data-grid/editing/IsCellEditableGrid.js", "bg": "inline"}}
+
+### Control editing
+
+The `editRowsModel` lets you control the editing.
+
+- implement the `onEditRowModelChange` to control the state of the `GridEditRowsModel` model.
+
+{{"demo": "pages/components/data-grid/editing/EditRowModelControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### value Getter
+
+If the column definition process their data using a `valueGetter`.
+
+- set the underlying data by implementing `onEditCellChangeCommitted` event handler invoked when the change is submitted, and the grid is about to update.
+
+{{"demo": "pages/components/data-grid/editing/ValueGetterGrid.js", "bg": "inline"}}
+
+### Validation
+
+To validate cell values on input change.
+
+- set the event handler `onEditCellChange` invoked when a change is triggered by the edit input component.
+- set the event handler `onEditCellChangeCommitted` to validate when the change is committed.
```

Comment:
```suggestion
- Set the event handler `onEditCellChangeCommitted` to only persist a value when valid. It's invoked when an end-user requests a change to be committed.
```

Reply:
What you're saying is not the same, I mean we can also do validation at this stage

Reply:
I was confused by "to validate when the change is committed". "validate" made me think of format validation, which it seems to be about when the user requests to save the changes, hence why I have used "persist".

Reply:
Set the event handler `onEditCellChangeCommitted` to validate or persist the new value. This handler is invoked when an end-user requests a change to be committed and before the cell revert back to view mode.